src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -104,9 +113,25 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     throw;
                 }
+                finally
+                {
+                    telemetryClients?.TrackEvent(telemetryEventDetails, TelemetryConstants.AcquireTokenEventName);
```

Comment:
>telemetryClients?.TrackEvent(telemetryEventDetails, TelemetryConstants.AcquireTokenEventName); [](http://example.com/codeflow?start=20&length=94)

I would think you'd only want to call TrackEvent IIF at least one telemetry client was enabled for the given event. Take a look how SAL does this (first checking if there are any enabled clients then, only for those clients passing the event along to it.)